Makefile

```diff
@@ -17,6 +17,8 @@ NODE_MODULES_TEST=package-lock.json
 WEBPACK_TEST=dist/manifest.json
 # one example file in src/lib to check if it was already checked out
 LIB_TEST=src/lib/cockpit-po-plugin.js
+# common arguments for tar, mostly to make the generated tarballs reproducible
+TAR_ARGS = --sort=name --mtime "@$(shell git show --no-patch --format='%at')" --mode=go=rX,u+rw,a-s --numeric-owner --owner=0 --group=0
```

Comment:
I was a bit concerned about this being at the top of the file, as the Makefile is also relevant for running in release tarballs (where there is no git dir).

But it works for the basic cases:
```
❱❱❱ make dist
❱❱❱ tar -C /tmp/ -xf cockpit-starter-kit-1.tar.xz
❱❱❱ make -C /tmp/cockpit-starter-kit/ bots
make: Entering directory '/tmp/cockpit-starter-kit'
git clone --quiet --reference-if-able ${XDG_CACHE_HOME:-$HOME/.cache}/cockpit-project/bots https://github.com/cockpit-project/bots.git
info: Could not add alternate for '/var/home/martin/.cache/cockpit-project/bots': path '/var/home/martin/.cache/cockpit-project/bots' does not exist
if [ -n "$COCKPIT_BOTS_REF" ]; then git -C bots fetch --quiet --depth=1 origin "$COCKPIT_BOTS_REF"; git -C bots checkout --quiet FETCH_HEAD; fi
checked out bots/ ref 3930f8dfef4ab7454b3d45ca943087a5fe09e8c7
make: Leaving directory '/tmp/cockpit-starter-kit'
```

However, it does fail if it somehow involves rebuilding the tarball:
```
❱❱❱ make -C /tmp/cockpit-starter-kit/ rpm
make: Entering directory '/tmp/cockpit-starter-kit'
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
if type appstream-util >/dev/null 2>&1; then appstream-util validate-relax --nonet *.metainfo.xml; fi
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
tar --xz --sort=name --mtime "@" --mode=go=rX,u+rw,a-s --numeric-owner --owner=0 --group=0 -cf cockpit-starter-kit-node-1.tar.xz node_modules
tar: Substituting -9223372036854775807 for unknown date format ‘@’
tar: node_modules: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
make: *** [Makefile:112: cockpit-starter-kit-node-1.tar.xz] Error 2
make: *** Waiting for unfinished jobs....
org.cockpit-project.starter-kit.metainfo.xml: OK
touch -r package.json package-lock.json
touch dist/*
tar --xz --sort=name --mtime "@" --mode=go=rX,u+rw,a-s --numeric-owner --owner=0 --group=0 -cf cockpit-starter-kit-1.tar.xz --transform 's,^,cockpit-starter-kit/,' \
	--exclude packaging/cockpit-starter-kit.spec.in --exclude node_modules \
	$(git ls-files) src/lib package-lock.json cockpit-starter-kit.spec dist/
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
tar: Substituting -9223372036854775807 for unknown date format ‘@’
make: Leaving directory '/tmp/cockpit-starter-kit'
```

However, this also fails on main, as that already tries to prod git for information.

So this seems good enough.